lib/utapi/utilities.js

```diff
@@ -280,11 +281,13 @@ function pushMetric(action, log, metricObj) {
             sizeDelta = -oldByteLength;
         } else if (action === 'abortMultipartUpload' && byteLength) {
             sizeDelta = -byteLength;
+        } else if (action === 'putDeleteMarkerObject' && versioningSuspended && byteLength) {
+            sizeDelta = -byteLength;
         }
 
         let objectDelta = isDelete ? -numberOfObjects : numberOfObjects;
         // putDeleteMarkerObject does not pass numberOfObjects
-        if (action === 'putDeleteMarkerObject'
+        if ((action === 'putDeleteMarkerObject' && !versioningSuspended)
```

Comment:
Shouldn't we also check `byteLength`? If it is used to know whether we are removing an object or not in the versioning-suspended case (in the sense that if we are not removing anything, we should still increment `objectDelta`)

Reply:
I believe you're right. If I bring `byteLength` into the picture I think I can drop the check of `versioningSuspended`, and just use `byteLength !== null` to cover all the cases.